packages/app/src/__tests__/CorePack.test.tsx

```diff
@@ -31,101 +29,105 @@ test('visualise scalar dataset', async () => {
-  ).resolves.toBeVisible();
+  await expect(findVisTabs()).resolves.toEqual([
+    Vis.Matrix,
+    Vis.Line,
+    Vis.Heatmap,
+  ]);
+  await expect(findSelectedVisTab()).resolves.toBe(Vis.Heatmap);
 
-  // Let the 2D slice fetch succeed
-  jest.runAllTimers();
-  await expect(screen.findByRole('figure')).resolves.toBeVisible();
+  const figure = await screen.findByRole('figure', { name: 'twoD' });
+  expect(figure).toBeVisible();
+  expect(within(figure).getByText('4e+2')).toBeVisible(); // color bar limit
+});
+
+test('visualize 1D slice of a 3D dataset as Line with and without autoscale', async () => {
+  jest.useFakeTimers();
+  const { user } = await renderApp({
+    initialPath: '/resilience/slow_slicing',
+    preferredVis: Vis.Line,
+  });
 
-  // Select the LineVis. The autoscale is on by default: it should fetch a 1D slice.
-  await user.click(await screen.findByRole('tab', { name: Vis.Line }));
+  // Wait for slice loader to appear (since autoscale is on by default, only the first slice gets fetched)
   await expect(
     screen.findByText(/Loading current slice/)
   ).resolves.toBeVisible();
 
-  // Let the 1D slice fetch succeed
+  // Wait for fetch of first slice to succeed
   jest.runAllTimers();
   await expect(screen.findByRole('figure')).resolves.toBeVisible();
 
-  // Check that autoscale is truly on
-  const autoScaleBtn = await screen.findByRole('button', {
-    name: 'Auto-scale',
-  });
+  // Confirm that autoscale is indeed on
+  const autoScaleBtn = screen.getByRole('button', { name: 'Auto-scale' });
   expect(autoScaleBtn).toHaveAttribute('aria-pressed', 'true');
 
-  // Move to other slice to fetch new slice
+  // Move to next slice
   const d0Slider = screen.getByRole('slider', { name: 'D0' });
   await user.type(d0Slider, '{ArrowUp}');
+
+  // Wait for slice loader to re-appear after debounce
   await expect(
     screen.findByText(/Loading current slice/)
   ).resolves.toBeVisible();
 
-  // Let the new slice fetch succeed
+  // Wait for fetch of second slice to succeed
   jest.runAllTimers();
   await expect(screen.findByRole('figure')).resolves.toBeVisible();
 
-  // Activate autoscale. It should trigger the fetch of the entire dataset.
+  // Activate autoscale
   await user.click(autoScaleBtn);
+
+  // Now, the entire dataset gets fetched
   await expect(
     screen.findByText(/Loading entire dataset/)
   ).resolves.toBeVisible();
 
-  // Let the dataset fetch succeed
+  // Wait for fetch of entire dataset to succeed
   jest.runAllTimers();
   await expect(screen.findByRole('figure')).resolves.toBeVisible();
 
-  // Move to third slice to check that entire dataset is fetched
+  // Move to third slice
   await user.type(d0Slider, '{ArrowUp}');
 
-  await expect(screen.findByRole('figure')).resolves.toBeVisible();
+  // Wait for new slicing to apply to Line visualization to confirm that no more slow fetching is performed
+  await expect(screen.findByTestId('2,0,x', undefined)).resolves.toBeVisible();
```

Comment:
Could you elaborate on the reason of this change and the need for a `testid` ?

Reply:
Ah yeah, so basically we don't have a way in the DOM to know when the visualization component has actually "applied" the new mapping/slicing. The title doesn't change, the color bar bounds may or may not change, etc. Since there is a debounce, we also can't count on the fact that the visualization is updated right away, so the `testid` gives us a way to wait for the visualization to update, which guarantees that the test has completed and that React won't perform any more work after that.

Reply:
In principle, I'd agree with you, but in this case, the information is really not easy to convey to a user by text (unless you had something in mind), and it really is the mapping/slicing information that we need to check, because it's the thing that changes. Adding this information to a caption/title would also duplicate information that is already present on the UI, thus actually _reducing_ accessibility (by slowing down screen reader users or adding clutter to the UI for sighted users).

I think in this case, `testid` (and the comment) is what keeps the test clear/concise/readable and also reliable. It makes the underlying testing complexity explicit instead of trying to hide it behind a workaround.

Reply:
Fair point. Let's keep the `testid` then :+1: